src/pumps/batch.rs

```diff
@@ -0,0 +1,73 @@
+use tokio::{
+    sync::mpsc::{self, Receiver},
+    task::JoinHandle,
+};
+
+use crate::Pump;
+
+pub struct BatchPump<T> {
+    pub(crate) n: usize,
+    batch: Vec<T>,
+}
+
+impl<T> BatchPump<T> {
+    pub fn new(n: usize) -> Self {
+        Self {
+            n,
+            batch: Vec::with_capacity(n),
+        }
+    }
+}
+
+impl<T> Pump<T, Vec<T>> for BatchPump<T>
+where
+    T: Send + 'static,
+{
+    fn spawn(mut self, mut input_receiver: Receiver<T>) -> (Receiver<Vec<T>>, JoinHandle<()>) {
+        let (output_sender, output_receiver) = mpsc::channel(1);
+
+        let h = tokio::spawn(async move {
+            while let Some(input) = input_receiver.recv().await {
+                self.batch.push(input);
+                if self.batch.len() >= self.n {
```

Comment:
why `>=` and not `==` ?

Reply:
It was for covering the idiotic case of `n==0`, but `==` is better anyway.